Task: ver.2.0 Docker in SpringBoot. Add Docker to the project. Executor: Oleksii Bashkirov.

OleksiiBashkirov opened the pull request:
Reason for Creating a New Pull Request:

In response to the feedback on my previous pull request(https://github.com/OleksiiBashkirov/online-book-store/pull/15), where it was noted that there were changes not directly related to the Docker task, I have taken the following steps to ensure clarity and focus on Docker integration:

Cloned the Main Branch: I cloned the latest code from the main branch to ensure that my work is based on the most up-to-date and stable version of the project.

Created a New Branch (add-docker-2): This new branch was specifically created to focus solely on integrating Docker into the project, avoiding any unrelated changes.

Added Only Docker-Related Files:

.env File: To manage environment variables necessary for Docker setup.
Dockerfile: To define the application image for Docker.
docker-compose.yml: To orchestrate Docker containers for the application.
spring-boot-docker-compose Dependency: Added to pom.xml to support Docker integration.

These actions were taken to maintain a clean and focused approach, ensuring that this pull request only includes changes necessary for Docker support. 
Welcome to any further comments or suggestions.

OleksiiBashkirov committed:
feat: Integrate Docker into the project

- Added .env file to manage environment variables for Docker setup
- Created Dockerfile for building the application image
- Added docker-compose.yml for orchestrating Docker containers
- Included spring-boot-docker-compose dependency in pom.xml for enhanced Docker support

okuzan commented on docker-compose.yml:
```diff
@@ -0,0 +1,32 @@
+      SPRING_DATASOURCE_USERNAME: ${MYSQLDB_USER}
+      SPRING_DATASOURCE_PASSWORD: ${MYSQLDB_ROOT_PASSWORD}
+      JWT_SECRET: ${JWT_SECRET}
+      JAVA_TOOL_OPTIONS: "-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=*:5005"
```
replace 5005 with DEBUG_PORT

okuzan reviewed (changes requested):
add .env to gitignore
create .env.template with actual values removed 
remove it from here while keeping locally with git rm --cached .env

OleksiiBashkirov committed:
feat: made corrections

- Removed .env file from version control and added it to .gitignore to protect sensitive information
- Created .env.template with placeholders for environment variables, enabling secure and easy configuration
- Updated docker-compose.yml to use environment variables for configuration, improving flexibility
- Replaced hardcoded debug port with DEBUG_PORT environment variable

OleksiiBashkirov committed:
- Also added .gitignore

OleksiiBashkirov committed:
- removed .env file again(
